converter/data_converter.go

```diff
@@ -52,4 +52,19 @@ type (
 		// ToStrings converts payloads object into human readable strings.
 		ToStrings(input *commonpb.Payloads) []string
 	}
+
+	// Stateful DataConverters can return a copy of themselves tailored to a Workflow or Activity context.
+	// This interface is optional.
```

Comment:
Maybe make it more explicit that this interface can only be used as an extension interface to a DataConverter implementation?